### PR TITLE
purestorage: deprecate leftovers

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY

Modules supporting purestorage were removed from the collection in version 3.0.0 but the module utils and a doc fragment were left behind. This PR deprecates them.

(Additionally, this PR fixes the ordering for the old rackspace removal in those two sections of runtime.yml)

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

plugins/doc_fragments/purestorage.py
plugins/module_utils/pure.py

### CODE REVIEW

File: changelogs/fragments/9432-deprecate-pure.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+deprecated_features:
+  - purestorage doc_fragments - modules using this were removed in community.general 3.0.0 (https://github.com/ansible-collections/community.general/pull/9432).
```
Comment (felixfontein) [inline changelogs/fragments/9432-deprecate-pure.yml:2]:
```suggestion
  - purestorage doc fragments - the doc fragment is deprecated and will be removed from community.general 12.0.0. The modules using this were removed in community.general 3.0.0 (https://github.com/ansible-collections/community.general/pull/9432).
```

File: changelogs/fragments/9432-deprecate-pure.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+  - pure module_utils - modules using this were removed in community.general 3.0.0 (https://github.com/ansible-collections/community.general/pull/9432).
```
Comment (felixfontein) [inline changelogs/fragments/9432-deprecate-pure.yml:3]:
```suggestion
  - pure module utils - the module utils is deprecated and will be removed from community.general 12.0.0. The modules using this were removed in community.general 3.0.0 (https://github.com/ansible-collections/community.general/pull/9432).
```

File: meta/runtime.yml [hunk 3]
```diff
@@ -827,11 +822,16 @@ plugin_routing:
       redirect: infoblox.nios_modules.nios
     postgresql:
       redirect: community.postgresql.postgresql
-  module_utils:
-    rax:
+    purestorage:
+      deprecation:
+      removal_version: 12.0.0
+      warning_text: The modules for purestorage were removed in community.general 3.0.0, this document fragment was left behind.
```
Comment (felixfontein) [inline meta/runtime.yml:828]:
```suggestion
        removal_version: 12.0.0
        warning_text: The modules for purestorage were removed in community.general 3.0.0, this document fragment was left behind.
```

File: meta/runtime.yml [hunk 4]
```diff
@@ -850,6 +850,14 @@ plugin_routing:
       redirect: infoblox.nios_modules.api
     postgresql:
       redirect: community.postgresql.postgresql
+    pure:
+      deprecation:
+      removal_version: 12.0.0
+      warning_text: The modules for purestorage were removed in community.general 3.0.0, this module util was left behind.
```
Comment (felixfontein) [inline meta/runtime.yml:856]:
```suggestion
        removal_version: 12.0.0
        warning_text: The modules for purestorage were removed in community.general 3.0.0, this module util was left behind.
```

**Comment by russoz** [comment]
Actually, the other PR just removed the doc fragment, so this one should be only abou the mod util

**Comment by felixfontein** [comment]
The doc fragment is still there, it has not been removed in the other PR.

File: changelogs/fragments/9432-deprecate-pure.yml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+deprecated_features:
+  - pure module_utils - modules using this were removed in community.general 3.0.0 (https://github.com/ansible-collections/community.general/pull/9432).
```
Comment (felixfontein) [inline changelogs/fragments/9432-deprecate-pure.yml:2]: I guess this should now be changed to be about the doc fragment.

Comment (russoz) [reply changelogs/fragments/9432-deprecate-pure.yml:2]: You lost me. The mod utils is still being deprecated, that has not changed, but #9433 has just removed the doc fragment, so how should this PR be about the doc fragment that no longer exists?

Comment (felixfontein) [reply changelogs/fragments/9432-deprecate-pure.yml:2]: The file plugins/doc_fragments/purestorage.py is still there, so it has not been removed and has to be deprecated before we can delete it.

Comment (felixfontein) [reply changelogs/fragments/9432-deprecate-pure.yml:2]: (The only thing that has been removed is the DOCUMENTATION fragment in plugins/doc_fragments/purestorage.py. The other fragments in there are still around.)

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks!

**Pull request merged**